spyder_vim/vim_widget.py

```diff
@@ -79,14 +89,19 @@ def DOLLAR(self, repeat=1):
     def ZERO(self, repeat=1):
         self._move_cursor(QTextCursor.StartOfLine)
 
-    def G(self, repeat):
-        editor = self._widget.editor()
-        editor.go_to_line(repeat)
-        self._widget.update_vim_cursor()
-
+    def G(self, repeat=1):
+        self._move_cursor(QTextCursor.Start)
```

Comment:
`number+G` is supposed to send the cursor to a specific line number, your version doesn't do that anymore ?
`G` puts the cursor to the top inly if no number was entered, i.e. `repeat == 1`